src/ncp/ncp_spinel.cpp

```diff
@@ -181,6 +183,7 @@ void NcpSpinel::HandleResponse(spinel_tid_t aTid, const uint8_t *aFrame, uint16_
         SuccessOrExit(error = SpinelDataUnpack(data, len, SPINEL_DATATYPE_UINT_PACKED_S, &spinelRole));
 
         mDeviceRole = SpinelRoleToDeviceRole(spinelRole);
+        mPropsObserver->SetDeviceRole(mDeviceRole);
         CallAndClear(mGetDeviceRoleHandler, OtbrErrorToOtError(error), mDeviceRole);
```

Comment:
What happens if there are multiple simultaneous call of `GetDeviceRole()`? It looks like a single `mGetDeviceRoleHandler` is recorded and invoked here. We don't support multiple calls to `GetDeviceRole()`?

Reply:
As discussed, let's remove the async version of `GetDeviceRole` and only use the sync version. The sync version returns immediately so can support multiple calls.

Reply:
Have removed the async version of `GetDeviceRole` and changed to use the `GetDeviceRole` provided by network properties.